testing/acceptance_test.py

```diff
@@ -1286,6 +1286,93 @@ def test_3():
     result.assert_outcomes(failed=3)
 
 
+def test_error_on_sync_test_async_fixture(pytester: Pytester) -> None:
+    pytester.makepyfile(
+        test_sync="""
+            import pytest
+
+            @pytest.fixture
+            async def async_fixture():
+                ...
+
+            def test_foo(async_fixture):
+                ...
+        """
+    )
+    result = pytester.runpytest()
+    result.stdout.fnmatch_lines(
+        [
+            (
+                "*ERROR: Sync test requested async fixture. "
```

Comment:
As I commented, I think improving the message here is important:

```suggestion
                "*ERROR: Sync test 'test_foo' requested async fixture 'async_fixture'. "
```